source/windows/environment.c

```diff
@@ -23,12 +23,16 @@ int aws_get_environment_value(
     const struct aws_string *variable_name,
     struct aws_string **value_out) {
 
-#pragma warning(push)
-#pragma warning(disable : 4996)
+#ifndef __MINGW32__
+#   pragma warning(push)
+#   pragma warning(disable : 4996)
```

Comment:
please comment what warning this disables (and say why if it's not immediately obvious why one would disable it)

For example, this is how we usually disable some MSVC-specific warnings
```
#if _MSC_VER
#    pragma warning(disable : 4221) /* aggregate initializer using local variable addresses */
#    pragma warning(disable : 4204) /* non-constant aggregate initializer */
#endif
```

Reply:
This (and the same construct in `system_info.c`) only looks like it's a newly disabled warning because of indentation changes. It's actually just disabling the MSVC specific #pragma when using MinGW as GCC can't understand MSVC warning codes.